raft/storage.go

```diff
@@ -130,6 +134,9 @@ func (ms *MemoryStorage) Term(i uint64) (uint64, error) {
 	if i < offset {
 		return 0, ErrCompacted
 	}
+	if int(i-offset) >= len(ms.ents) {
+		return 0, ErrTermUnavailable
```

Comment:
let's return ErrUnavailable instead. it is because the entry with the requested index is not available yet.